activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -22,6 +21,15 @@ def stub_object(object, method_name, return_value)
         object.define_singleton_method(method_name) { return_value }
       end
 
+      def unstub(object, method_name)
+        key = [object.object_id, method_name]
+        unstub_object(@stubs.delete(key))
+
+        if old_stub = @pre_stubs.fetch(key, Array.new).pop
```

Comment:
I find this very hard to understand. Maybe extract this line into a method to clarify its intent?